### PR TITLE
[fix](multi-catalog)put java udf to custom lib

### PR DESCRIPTION
**Pull request by wsjz**
## Proposed changes

when use java udf , we can put its jars in custom_lib of FE and BE in advance， and then we can create function without file property:
```
CREATE FUNCTION java_udf_add_one(int) RETURNS int PROPERTIES (
    "symbol"="org.apache.doris.udf.AddOne",
    "always_nullable"="true",
    "type"="JAVA_UDF"
);
```

and we can use file property for custom jar:  

```
CREATE FUNCTION java_udf_add_one(int) RETURNS int PROPERTIES (
    "file"="file:///path/to/java-udf-demo.jar",
    "symbol"="org.apache.doris.udf.AddOne",
    "always_nullable"="true",
    "type"="JAVA_UDF"
);
```

on FE, we use null location if the 'file' property is not set
on BE, we get jar path if both file path location and checksum are null

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by hello-stephen** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

File: run-regression-test.sh [hunk 1]
```diff
@@ -178,6 +182,11 @@ if ! test -f ${RUN_JAR:+${RUN_JAR}}; then
     cd "${DORIS_HOME}"/regression-test/java-udf-src
```
Comment (hello-stephen) [inline run-regression-test.sh:182]: --compile时上面161行退出了，需要改下

**Comment by hello-stephen** [comment]
run p0

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by BePPPower — APPROVED** [review]

**Pull request merged**